src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -969,10 +969,25 @@ because xbuild doesn't support framework reference assemblies.
 />
 
 <Target Name="UpdateAndroidResources"
-    Condition=" '$(DesignTimeBuild)' != 'True' Or '$(DeferredBuildSupported)' != 'True' Or '$(DeferredBuild)' == 'True' "
+    Condition=" ('$(DesignTimeBuild)' != 'True' Or '$(DeferredBuildSupported)' != 'True' Or '$(DeferredBuild)' == 'True') And !Exists ('$(_AndroidStampDirectory)_UpdateAndroidResources.stamp')"
     DependsOnTargets="_SetupDesignTimeBuildForCompile;_ManagedUpdateAndroidResgen;_UpdateAndroidResources"
 />
 
+<Target Name="_BeforeUpdateAndroidResources"
+    BeforeTargets="_SetupDesignTimeBuildForCompile"
+    Condition="!Exists ('$(_AndroidStampDirectory)_UpdateAndroidResources.stamp')"
+>
+  <Touch Files="$(_AndroidStampDirectory)_UpdateAndroidResources.stamp" AlwaysCreate="True" />
+  <ItemGroup>
+    <FileWrites Include="$(_AndroidStampDirectory)_UpdateAndroidResources.stamp" />
+  </ItemGroup>
+</Target>
+
+<Target Name="_AfterUpdateAndroidResources" AfterTargets="UpdateAndroidResources">
```

Comment:
Does an "AfterTargets" execute if the specified target doesn't execute?

`UpdateAndroidResources` is updated with a `Condition` so that it "doesn't run" if `_UpdateAndroidResources.stamp` exists.  Thus, if `UpdateAndroidResources` is currently executing -- and thus `_UpdateAndroidResources.stamp` exists -- and then a separate process/whatever *also* runs the `UpdateAndroidResources` target, which is a no-op because `_UpdateAndroidResources.stamp`, so the 2nd `UpdateAndroidResource` doesn't run -- will the separate "no-op" process _also_ run the `_AfterUpdateAndroidResources` target, thus deleting `_UpdateAndroidResources.stamp`, which will prevent this "file lock" solution from scaling to a *third* process running `UpdateAndroidResources`?

We might need to have "per-invocation" stamp files, and have `_AfterUpdateAndroidResources` *only* delete the file it created.  (Which might be differently "weird" if someone *kills* the msbuild process while it's running…?)

Reply:
Yup this was my concern. But we also need to handle the situation where the underlying Aapt2 call Errors, in which case none of the dependencies will get run unless we use AfterTargets. So I'm not 100% sure this is going to work.